web-components/src/components/card/Card.ts

```diff
@@ -1,53 +1,153 @@
+import "@/components/list/List";
+import "@/components/list/ListItem";
+import "@/components/favorite/Favorite";
+import { repeat } from "lit-html/directives/repeat";
+import { classMap } from "lit-html/directives/class-map";
+import { nothing } from "lit-html";
 
 export namespace Card {
   @customElementWithCheck("md-card")
   export class ELEMENT extends LitElement {
     @property({ type: String }) title = "title";
     @property({ type: String }) subtitle = "";
-    @property({ type: String }) src = "";
-    @property({ type: String }) link = "";
-    @property({ type: String }) linkTitle = "";
-    @property({ type: String }) buttonTitle = "";
+    @property({ type: String }) id = "";
+    @property({ type: Boolean }) fullscreen = false;
+    @property({ type: String }) info = "";
+    @property({ type: Array }) menuOption: (string | { [key: string]: string })[] = [];
+
+    @internalProperty() private full = false;
 
     static get styles() {
       return [reset, styles];
     }
 
+    get cardClassMap() {
+      return {
+        "full-screen": this.full
+      };
+    }
+
+    connectedCallback() {
+      super.connectedCallback();
+    }
+
+    handleToggleExpandCollapse() {
+      this.full = !this.full;
+    }
+
+    handleCardMenuEvent(event: MouseEvent, label: string) {
+      event.preventDefault();
+      const ev = label.toLowerCase();
+      this.dispatchEvent(
+        new CustomEvent<{ id: string; type: string }>("card-click", {
+          detail: {
+            id: this.id,
+            type: ev
+          },
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+
+    handleCardKeyDownEvent(event: KeyboardEvent, label: string) {
+      this.dispatchEvent(
+        new CustomEvent<{ id: string; type: string }>("card-keydown", {
+          detail: {
+            id: this.id,
+            type: label.toLowerCase()
+          },
+          bubbles: true,
+          composed: true
+        })
+      );
+    }
+
     render() {
       return html`
-        <div class="md-card" part="card">
+        <div class="md-card ${classMap(this.cardClassMap)}" id="${this.id}" part="card">
           <div class="md-card-header">
-            <slot name="header">
-              <md-avatar alt="avatar" label="avatar" title=${this.title} src=${this.src}></md-avatar>
-              <div class="md-card-header-title">
-                <h3>${this.title}</h3>
-                <h5>${this.subtitle}</h5>
-              </div>
-              <div class="md-card-header-actions">
-                <md-menu-overlay placement="bottom-end">
-                  <md-button variant="white" slot="menu-trigger" circle>
-                    <md-icon slot="icon" name="icon-more-adr_16"></md-icon>
-                  </md-button>
-                  <md-list></md-list>
-                </md-menu-overlay>
-              </div>
+            <slot name="card-header-icon">
+              <md-favorite value="${`card-` + this.id}"></md-favorite>
```

Comment:
it is not really an icon slot. it can be anything: an icon, a favorite button (button is not just the icon. it is an action) an avatar. an image...
what can we name it to be generic enough?.. `card-header-aside`

Reply:
Renamed